src/job.ts

```diff
@@ -248,24 +251,20 @@ export class Job {
         }
 
         if (this.image) {
-            const time = process.hrtime();
-            let preCmd = ``;
-            preCmd += `docker run --rm -v $PWD:/app/ -w /app/ eeacms/rsync sh -c "\n`;
-            preCmd += `mkdir -p .gitlab-ci-local/builds/${safeName}\n`
-            preCmd += `rsync -a . .gitlab-ci-local/builds/${safeName}/. --delete --exclude '.gitlab-ci-local/'\n`
-            preCmd += `chmod a+w -R .gitlab-ci-local/builds/${safeName}\n`
-            preCmd += `chown root:root -R .gitlab-ci-local/builds/${safeName}\n`
-            if (fs.existsSync(`${this.cwd}/.gitlab-ci-local/file-variables/`)) {
-                preCmd += `mkdir -p .gitlab-ci-local/builds/${safeName}/.gitlab-ci-local/file-variables/\n`;
-                preCmd += `rsync -a .gitlab-ci-local/file-variables/. .gitlab-ci-local/builds/${safeName}/.gitlab-ci-local/file-variables/.\n`;
-            }
-            preCmd += `"\n`;
-            await Utils.spawn(preCmd, this.cwd);
-            const endTime = process.hrtime(time);
-            process.stdout.write(`${this.getJobNameString()} ${magentaBright(`rsync to build folder`)} in ${magenta(prettyHrtime(endTime))}\n`);
+            time = process.hrtime();
+            process.stdout.write(`${jobNameStr} ${magentaBright(`pulling`)} ${this.image}\n`);
+            let pullCmd = ``;
+            pullCmd += `docker image ls --format '{{.Repository}}:{{.Tag}}' | grep -E '^${this.image}$'\n`
+            pullCmd += `if [ "$?" -ne 0 ]; then\n`
```

Comment:
wouldn't this look better as one expression?

Reply:
Sure, if we don't care about line length 😃
And I don't so will change

Reply:
Wait, we cannot put all this on one line it will be way too long

Reply:
```
            pullCmd += `docker image ls --format '{{.Repository}}:{{.Tag}}' | grep -E '^${this.image}$'\n`
            pullCmd += `if [ "$?" -ne 0 ]; then\n`
            pullCmd += `\techo "Pulling ${this.image}"\n`
            pullCmd += `\tdocker pull ${this.image}\n`
            pullCmd += `fi\n`
```